### ISSUE TITLE
Replication suddenly stops, code 503

### ISSUE THREAD

**Issue by ltamrazov** [couchbaselabs/elasticsearch-transport-couchbase#70]
Hi, 

We start up replication, (which just a few documents). Everything gets replicated correctly and is indexed nicely. Then at some point (usually within half hour or so) we get replication errors (even though no new documents have been added) and from that point onward, if we try to add new documents, they do not get replicated. 

At first the replication is successful, and all the documents get replicated ( so it is not the case with invalid JSON). But after a while it crashes without us having made any changes to the dataset. 

I've checked that couchbaseCheckpoint documents are present, and each contains a source with uuid. The structure of each is as follows:
```
    {
        _index: main_v1
        _type: couchbaseCheckpoint
        _id: vbucket3UUID 
        _version: 1
        _score: 1
        _source: {
            doc: {
                   uuid: 49525f6d9e6e401a968f696aa1c9f974
            }
        }
    }
```
Elastic logs are clean. Couchbase logs keep printing the following:

couchdb.log and xdcr_errors.log both print this:

```
[couchdb:info,2015-02-12T19:24:57.385,ns_1@couchbase.dev.mydomain.com:   <0.10441.10>:couch_log:info:41]Retrying POST request to http://Administrator:*****@192.168.1.100:9091/main_v1%2f18/_bulk_docs in 0.5 seconds due to error {code,503}
[couchdb:info,2015-02-12T19:24:57.512,ns_1@couchbase.dev.mydomain.com:<0.10459.10>:couch_log:info:41]Retrying POST request to http://Administrator:*****@192.168.1.100:9091/main_v1%2f60/_revs_diff in 0.5 seconds due to error {code,503}
[couchdb:error,2015-02-12T19:24:57.890,ns_1@couchbase.dev.mydomain.com:<0.10441.10>:couch_log:error:44]Replicator, request POST to "http://Administrator:*****@192.168.1.100:9091/main_v1%2f18/_bulk_docs" failed due to error {code,503}
[couchdb:info,2015-02-12T19:24:57.901,ns_1@couchbase.dev.mydomain.com:<0.10479.10>:couch_log:info:41]Retrying POST request to http://Administrator:*****@192.168.1.100:9091/main_v1%2f18/_bulk_docs in 0.25 seconds due to error {code,503}
[couchdb:error,2015-02-12T19:24:58.018,ns_1@couchbase.dev.mydomain.com:<0.10459.10>:couch_log:error:44]Replicator, request POST to "http://Administrator:*****@192.168.1.100:9091/main_v1%2f60/_revs_diff" failed due to error {code,503}
[couchdb:info,2015-02-12T19:24:58.032,ns_1@couchbase.dev.mydomain.com:<0.10497.10>:couch_log:info:41]Retrying POST request to http://Administrator:*****@192.168.1.100:9091/main_v1%2f60/_revs_diff in 0.25 seconds due to error {code,503}
[couchdb:info,2015-02-12T19:24:58.157,ns_1@couchbase.dev.mydomain.com:<0.10479.10>:couch_log:info:41]Retrying POST request to http://Administrator:*****@192.168.1.100:9091/main_v1%2f18/_bulk_docs in 0.5 seconds due to error {code,503}
[couchdb:info,2015-02-12T19:24:58.285,ns_1@couchbase.dev.mydomain.com:<0.10497.10>:couch_log:info:41]Retrying POST request to http://Administrator:*****@192.168.1.100:9091/main_v1%2f60/_revs_diff in 0.5 seconds due to error {code,503}
```

debug.log prints the following:
```
[ns_server:debug,2015-02-12T20:24:01.961,ns_1@couchbase.dev.mydomain.com:<0.5409.14>:remote_clusters_info:do_mk_json_get:1527]Doing get of http://192.168.1.100:9091/pools/default/buckets?uuid=dc197740d72e32c7abc6b40b0890baf4&forXDCR=1
[ns_server:debug,2015-02-12T20:24:01.963,ns_1@couchbase.dev.mydomain.com:<0.5409.14>:remote_clusters_info:do_with_mcd_to_couch_uri_dict:1414]Going to pick plain couchApiBase for mcd_to_couch_uri_dict: <<"http://192.168.1.100:9091/main_v1">>
[ns_server:debug,2015-02-12T20:24:01.966,ns_1@couchbase.dev.mydomain.com:<0.5408.14>:xdc_vbucket_rep:init_replication_state:650]Inited replication position: {0,0,0,0,<<"be9367b897fb4e38aa7884e99ccf27f7">>}
[ns_server:debug,2015-02-12T20:24:02.728,ns_1@couchbase.dev.mydomain.com:remote_clusters_info<0.685.0>:remote_clusters_info:handle_call:369]FoundCluster: {remote_cluster,<<"dc197740d72e32c7abc6b40b0890baf4">>,
                  [{remote_node,"192.168.1.100",9091,9091,undefined,
                       undefined,undefined}],
                  undefined}
[ns_server:debug,2015-02-12T20:24:02.728,ns_1@couchbase.dev.mydomain.com:<0.5430.14>:remote_clusters_info:do_mk_json_get:1527]Doing get of http://192.168.1.100:9091/pools
[error_logger:error,2015-02-12T20:24:02.728,ns_1@couchbase.dev.mydomain.com:error_logger<0.6.0>:ale_error_logger_handler:do_log:203]** Generic server <0.5408.14> terminating 
** Last message in was {'EXIT',<0.5418.14>,
                           {http_request_failed,"POST",
                               "http://Administrator:*****@192.168.1.100:9091/main_v1%2f63/_bulk_docs",
                               {error,{code,503}}}}
** When Server state == [{data,
                          [{"State",
                            {rep_state,
                             {rep,
                              <<"dc197740d72e32c7abc6b40b0890baf4/main/main_v1">>,
                              <<"main">>,
                              <<"/remoteClusters/dc197740d72e32c7abc6b40b0890baf4/buckets/main_v1">>,
                              "capi",
                              [{optimistic_replication_threshold,256},
                               {worker_processes,4},
                               {worker_batch_size,500},
                               {failure_restart_interval,30},
                               {doc_batch_size_kb,2048},
                               {checkpoint_interval,1800},
                               {max_concurrent_reps,16},
                               {connection_timeout,180},
                               {http_connections,20},
                               {retries_per_request,2},
                               {socket_options,
                                [{keepalive,true},{nodelay,false}]},
                               {pause_requested,false},
                               {supervisor_max_r,25},
                               {supervisor_max_t,5}]},
                             {rep_vb_status,63,<0.5408.14>,replicating},
                             {1423,790641,966684},
                             nil,nil,<0.11052.10>,<0.11049.10>,<<"main/63">>,
                             <<"*****@192.168.1.100:9091/main_v1%2f63">>,
                             {httpdb,
                              "*****@192.168.1.100:9091/main_v1%2f63/",nil,
                              [{"Accept","application/json"},
                               {"User-Agent","CouchDB/2.1.1r-432-gc2af28d"}],
                              180000,
                              [{connect_options,
                                [{keepalive,true},{nodelay,false}]},
                               {connect_timeout,180000},
                               {send_retry,0}],
                              2,250,xdc_lhttpc_pool,20},
                             {db,<0.1830.0>,<0.1831.0>,nil,
                              <<"1423773839449534">>,<0.1826.0>,<0.1836.0>,
                              {db_header,11,0,nil,nil,
                               <<0,0,0,0,97,216,0,0,0,0,5,139>>,
                               0,nil,nil},
                              0,
                              {btree,<0.1826.0>,nil,
                               #Fun<couch_db_updater.7.78420415>,
                               #Fun<couch_db_updater.8.78420415>,
                               #Fun<couch_btree.1.39972947>,
                               #Fun<couch_db_updater.9.78420415>,1279,2558,
                               true},
                              {btree,<0.1826.0>,nil,
                               #Fun<couch_db_updater.10.78420415>,
                               #Fun<couch_db_updater.11.78420415>,
                               #Fun<couch_db_updater.6.78420415>,
                               #Fun<couch_db_updater.12.78420415>,1279,2558,
                               true},
                              {btree,<0.1826.0>,
                               {25048,<<>>,1419},
                               identity,identity,#Fun<couch_btree.1.39972947>,
                               nil,1279,2558,true},
                              0,<<"main/master">>,
                              "/Users/Levon/Library/Application Support/Couchbase/var/lib/couchdb/main/master.couch.1",
                              [],nil,
                              {user_ctx,null,[],undefined},
                              nil,
                              [before_header,after_header,on_file_open],
                              [create]},
                             <<"be9367b897fb4e38aa7884e99ccf27f7">>,0,
                             undefined,0,0,4230,4230,0,13797118081439,
                             [4230],
                             0,
                             {0,0},
                             "Fri, 13 Feb 2015 01:24:01 GMT",
                             {1423792441967640,#Ref<0.0.20.120150>},
                             {1423,790641,967308},
                             {1423,790641,966710},
                             [<0.5418.14>],
                             <0.5415.14>,false,
                             {"http://Administrator:kva%24im0d0@192.168.1.100:9091/",
                              [{<<"vb">>,63},
                               {<<"bucket">>,<<"main_v1">>},
                               {<<"bucketUUID">>,
                                <<"fc70a8224cc64abe933b826a29c8bba3">>}],
                              {httpdb,
                               "*****@192.168.1.100:9091/main_v1%2f63/",nil,
                               [{"Accept","application/json"},
                                {"User-Agent","CouchDB/2.1.1r-432-gc2af28d"}],
                               180000,
                               [{connect_options,
                                 [{keepalive,true},{nodelay,false}]},
                                {connect_timeout,180000},
                                {send_retry,0}],
                               2,250,xdc_lhttpc_pool,20}}}}]}]
** Reason for termination == 
** {http_request_failed,"POST",
                        "http://Administrator:*****@192.168.1.100:9091/main_v1%2f63/_bulk_docs",
                        {error,{code,503}}}

[ns_server:debug,2015-02-12T20:24:02.729,ns_1@couchbase.dev.mydomain.com:<0.5430.14>:remote_clusters_info:do_mk_json_get:1527]Doing get of http://192.168.1.100:9091/pools/default?uuid=dc197740d72e32c7abc6b40b0890baf4
[error_logger:error,2015-02-12T20:24:02.732,ns_1@couchbase.dev.mydomain.com:error_logger<0.6.0>:ale_error_logger_handler:do_log:203]
=========================CRASH REPORT=========================
  crasher:
    initial call: xdc_vbucket_rep:init/1
    pid: <0.5408.14>
    registered_name: []
    exception exit: {http_request_failed,"POST",
                        "http://Administrator:*****@192.168.1.100:9091/main_v1%2f63/_bulk_docs",
                        {error,{code,503}}}
      in function  gen_server:terminate/6 (gen_server.erl, line 744)
    ancestors: [<0.11053.10>,<0.11049.10>,xdc_replication_sup,xdcr_sup,
                  ns_server_sup,ns_server_cluster_sup,<0.59.0>]
    messages: []
    links: [<0.11053.10>]
    dictionary: [{changes_reader,<0.5416.14>}]
    trap_exit: true
    status: running
    heap_size: 6772
    stack_size: 27
    reductions: 13694
  neighbours:

[error_logger:info,2015-02-12T20:24:02.732,ns_1@couchbase.dev.mydomain.com:error_logger<0.6.0>:ale_error_logger_handler:do_log:203]
=========================PROGRESS REPORT=========================
          supervisor: {<0.11053.10>,xdc_vbucket_rep_sup}
             started: [{pid,<0.5429.14>},
                       {name,63},
                       {mfa,
                        {xdc_vbucket_rep,start_link,
                         [{rep,
                           <<"dc197740d72e32c7abc6b40b0890baf4/main/main_v1">>,
                           <<"main">>,
                           <<"/remoteClusters/dc197740d72e32c7abc6b40b0890baf4/buckets/main_v1">>,
                           "capi",
                           [{optimistic_replication_threshold,256},
                            {worker_processes,4},
                            {worker_batch_size,500},
                            {failure_restart_interval,30},
                            {doc_batch_size_kb,2048},
                            {checkpoint_interval,1800},
                            {max_concurrent_reps,16},
                            {connection_timeout,180},
                            {http_connections,20},
                            {retries_per_request,2},
                            {socket_options,
                             [{keepalive,true},{nodelay,false}]},
                            {pause_requested,false},
                            {supervisor_max_r,25},
                            {supervisor_max_t,5}]},
                          63,<0.11051.10>,<0.11052.10>,<0.11049.10>,"capi"]}},
                       {restart_type,{permanent,30}},
                       {shutdown,100},
                       {child_type,worker}]

[ns_server:debug,2015-02-12T20:24:02.732,ns_1@couchbase.dev.mydomain.com:<0.5430.14>:remote_clusters_info:do_mk_json_get:1527]Doing get of http://192.168.1.100:9091/pools/default/buckets?uuid=dc197740d72e32c7abc6b40b0890baf4&forXDCR=1
[ns_server:debug,2015-02-12T20:24:02.735,ns_1@couchbase.dev.mydomain.com:<0.5430.14>:remote_clusters_info:do_with_mcd_to_couch_uri_dict:1414]Going to pick plain couchApiBase for mcd_to_couch_uri_dict: <<"http://192.168.1.100:9091/main_v1">>
[ns_server:debug,2015-02-12T20:24:02.737,ns_1@couchbase.dev.mydomain.com:<0.5429.14>:xdc_vbucket_rep:init_replication_state:650]Inited replication position: {0,0,0,0,<<"be9367b897fb4e38aa7884e99ccf27f7">>}
 ```

Any guidance would be really appreciated. I could send over the full logs if that helps.

Thank you,

**Comment by Branor** [comment]
Yup that was a nasty little bug with an incorrect concurrent requests counter. At some point it caused the plugin to reject all requests because it (wrongly) reached the maximum allowed number of concurrent requests. For a fixed version, take a look here: https://github.com/Branor/elasticsearch-transport-couchbase/releases/tag/v2.1.0-SNAPSHOT

**Comment by ltamrazov** [comment]
Hi, 

Thank you for the reply and the fix. I implemented it, replication seems to work thus far, however we started getting this error from elastic:

```
[2015-02-16 11:03:54,511][WARN ][org.eclipse.jetty.servlet.ServletHandler] /main_v1%2f21/_bulk_docs
java.lang.RuntimeException: indexing error, bulk response was null
	at org.elasticsearch.transport.couchbase.capi.ElasticSearchCAPIBehavior.bulkDocs(ElasticSearchCAPIBehavior.java:541)
	at com.couchbase.capi.servlet.CAPIServlet.handleBulkDocs(CAPIServlet.java:529)
	at com.couchbase.capi.servlet.CAPIServlet.service(CAPIServlet.java:89)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:847)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:594)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:485)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:119)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:486)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:233)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1065)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:412)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:192)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:999)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:117)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:111)
	at org.eclipse.jetty.server.Server.handle(Server.java:351)
	at org.eclipse.jetty.server.AbstractHttpConnection.handleRequest(AbstractHttpConnection.java:451)
	at org.eclipse.jetty.server.AbstractHttpConnection$RequestHandler.content(AbstractHttpConnection.java:931)
	at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:857)
	at org.eclipse.jetty.http.HttpParser.parseAvailable(HttpParser.java:235)
	at org.eclipse.jetty.server.AsyncHttpConnection.handle(AsyncHttpConnection.java:76)
	at org.eclipse.jetty.io.nio.SelectChannelEndPoint.handle(SelectChannelEndPoint.java:609)
	at org.eclipse.jetty.io.nio.SelectChannelEndPoint$1.run(SelectChannelEndPoint.java:45)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:599)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:534)
	at java.lang.Thread.run(Thread.java:745)
```

It does not seem to stop replication (adding new documents after we get this error works fine). Also it only happens with a bucket that has documents we excluded from replication using the following config:

```
# filter out all docs that don't match this
couchbase.keyFilter: org.elasticsearch.transport.couchbase.capi.RegexKeyFilter
couchbase.keyFilter.type: "include"
couchbase.keyFilter.keyFiltersRegex.idsToInclude: ^[updbrv]{1}-.+$
```

Not sure if its relevant, but the config also uses a typeSelector with multiple regexes. 

```couchbase.typeSelector: org.elasticsearch.transport.couchbase.capi.RegexTypeSelector```

Again, I'm pointing this out because we are replicating two buckets. In one, all the ids get replicated. The other contains ids that get replicated as well as ids that we filter out using the regex (that's the one that produces the error). 

couchdb.log prints:

```
[couchdb:error,2015-02-16T11:57:18.683,ns_1@couchbase.dev.mydomain.com:<0.31801.2>:couch_log:error:44]Replicator, request POST to "http://Administrator:*****@192.168.1.100:9091/main_v1%2f27/_bulk_docs" failed due to error {code,500}
```

debug.log: 

```
[error_logger:error,2015-02-16T11:57:50.716,ns_1@couchbase.dev.mydomain.com:error_logger<0.6.0>:ale_error_logger_handler:do_log:203]** Generic server <0.32498.2> terminating 
** Last message in was {'EXIT',<0.32511.2>,
                           {http_request_failed,"POST",
                               "http://Administrator:*****@192.168.1.100:9091/main_v1%2f41/_bulk_docs",
                               {error,{code,500}}}}
** When Server state == [{data,
                          [{"State",
                            {rep_state,
                             {rep,
                              <<"dc197740d72e32c7abc6b40b0890baf4/main/main_v1">>,
                              <<"main">>,
                              <<"/remoteClusters/dc197740d72e32c7abc6b40b0890baf4/buckets/main_v1">>,
                              "capi",
                              [{optimistic_replication_threshold,256},
                               {worker_processes,4},
                               {worker_batch_size,500},
                               {failure_restart_interval,30},
                               {doc_batch_size_kb,2048},
                               {checkpoint_interval,1800},
                               {max_concurrent_reps,16},
                               {connection_timeout,180},
                               {http_connections,20},
                               {retries_per_request,2},
                               {socket_options,
                                [{keepalive,true},{nodelay,false}]},
                               {pause_requested,false},
                               {supervisor_max_r,25},
                               {supervisor_max_t,5}]},
                             {rep_vb_status,41,<0.32498.2>,replicating},
                             {1424,105869,949679},
                             nil,nil,<0.28879.0>,<0.28876.0>,<<"main/41">>,
                             <<"*****@192.168.1.100:9091/main_v1%2f41">>,
                             {httpdb,
                              "*****@192.168.1.100:9091/main_v1%2f41/",nil,
                              [{"Accept","application/json"},
                               {"User-Agent","CouchDB/2.1.1r-432-gc2af28d"}],
                              180000,
                              [{connect_options,
                                [{keepalive,true},{nodelay,false}]},
                               {connect_timeout,180000},
                               {send_retry,0}],
                              2,250,xdc_lhttpc_pool,20},
                             {db,<0.459.0>,<0.460.0>,nil,
                              <<"1424098902710912">>,<0.456.0>,<0.472.0>,
                              {db_header,11,0,nil,nil,
                               <<0,0,0,0,161,242,0,0,0,0,19,46>>,
                               1,nil,nil},
                              0,
                              {btree,<0.456.0>,nil,
                               #Fun<couch_db_updater.7.78420415>,
                               #Fun<couch_db_updater.8.78420415>,
                               #Fun<couch_btree.1.39972947>,
                               #Fun<couch_db_updater.9.78420415>,1279,2558,
                               true},
                              {btree,<0.456.0>,nil,
                               #Fun<couch_db_updater.10.78420415>,
                               #Fun<couch_db_updater.11.78420415>,
                               #Fun<couch_db_updater.6.78420415>,
                               #Fun<couch_db_updater.12.78420415>,1279,2558,
                               true},
                              {btree,<0.456.0>,
                               {41458,<<>>,4910},
                               identity,identity,#Fun<couch_btree.1.39972947>,
                               nil,1279,2558,true},
                              0,<<"main/master">>,
                              "/Users/Levon/Library/Application Support/Couchbase/var/lib/couchdb/main/master.couch.2",
                              [],nil,
                              {user_ctx,null,[],undefined},
                              nil,
                              [before_header,after_header,on_file_open],
                              []},
                             <<"93eb051635834804844f1069ef444bd7">>,0,
                             undefined,0,0,22,22,0,245176386652781,
                             [22],
                             0,
                             {0,0},
                             "Mon, 16 Feb 2015 16:57:49 GMT",
                             {1424107669950445,#Ref<0.0.6.72623>},
                             {1424,105869,950200},
                             {1424,105869,949700},
                             [<0.32511.2>],
                             <0.32508.2>,false,
                             {"http://Administrator:kva%24im0d0@192.168.1.100:9091/",
                              [{<<"vb">>,41},
                               {<<"bucket">>,<<"main_v1">>},
                               {<<"bucketUUID">>,
                                <<"baa2747c512c4817bc52feecde9687ea">>}],
                              {httpdb,
                               "*****@192.168.1.100:9091/main_v1%2f41/",nil,
                               [{"Accept","application/json"},
                                {"User-Agent","CouchDB/2.1.1r-432-gc2af28d"}],
                               180000,
                               [{connect_options,
                                 [{keepalive,true},{nodelay,false}]},
                                {connect_timeout,180000},
                                {send_retry,0}],
                               2,250,xdc_lhttpc_pool,20}}}}]}]
** Reason for termination == 
** {http_request_failed,"POST",
                        "http://Administrator:*****@192.168.1.100:9091/main_v1%2f41/_bulk_docs",
                        {error,{code,500}}}

[error_logger:error,2015-02-16T11:57:50.718,ns_1@couchbase.dev.mydomain.com:error_logger<0.6.0>:ale_error_logger_handler:do_log:203]
=========================CRASH REPORT=========================
  crasher:
    initial call: xdc_vbucket_rep:init/1
    pid: <0.32498.2>
    registered_name: []
    exception exit: {http_request_failed,"POST",
                        "http://Administrator:*****@192.168.1.100:9091/main_v1%2f41/_bulk_docs",
                        {error,{code,500}}}
      in function  gen_server:terminate/6 (gen_server.erl, line 744)
    ancestors: [<0.28880.0>,<0.28876.0>,xdc_replication_sup,xdcr_sup,
                  ns_server_sup,ns_server_cluster_sup,<0.59.0>]
    messages: []
    links: [<0.28880.0>]
    dictionary: [{changes_reader,<0.32509.2>}]
    trap_exit: true
    status: running
    heap_size: 4185
    stack_size: 27
    reductions: 13740
  neighbours:

[error_logger:error,2015-02-16T11:57:50.835,ns_1@couchbase.dev.mydomain.com:error_logger<0.6.0>:ale_error_logger_handler:do_log:203]** Generic server <0.32520.2> terminating 
** Last message in was {'EXIT',<0.32530.2>,
                           {http_request_failed,"POST",
                               "http://Administrator:*****@192.168.1.100:9091/main_v1%2f32/_bulk_docs",
                               {error,{code,500}}}}
** When Server state == [{data,
                          [{"State",
                            {rep_state,
                             {rep,
                              <<"dc197740d72e32c7abc6b40b0890baf4/main/main_v1">>,
                              <<"main">>,
                              <<"/remoteClusters/dc197740d72e32c7abc6b40b0890baf4/buckets/main_v1">>,
                              "capi",
                              [{optimistic_replication_threshold,256},
                               {worker_processes,4},
                               {worker_batch_size,500},
                               {failure_restart_interval,30},
                               {doc_batch_size_kb,2048},
                               {checkpoint_interval,1800},
                               {max_concurrent_reps,16},
                               {connection_timeout,180},
                               {http_connections,20},
                               {retries_per_request,2},
                               {socket_options,
                                [{keepalive,true},{nodelay,false}]},
                               {pause_requested,false},
                               {supervisor_max_r,25},
                               {supervisor_max_t,5}]},
                             {rep_vb_status,32,<0.32520.2>,replicating},
                             {1424,105870,66191},
                             nil,nil,<0.28879.0>,<0.28876.0>,<<"main/32">>,
                             <<"*****@192.168.1.100:9091/main_v1%2f32">>,
                             {httpdb,
                              "*****@192.168.1.100:9091/main_v1%2f32/",nil,
                              [{"Accept","application/json"},
                               {"User-Agent","CouchDB/2.1.1r-432-gc2af28d"}],
                              180000,
                              [{connect_options,
                                [{keepalive,true},{nodelay,false}]},
                               {connect_timeout,180000},
                               {send_retry,0}],
                              2,250,xdc_lhttpc_pool,20},
                             {db,<0.459.0>,<0.460.0>,nil,
                              <<"1424098902710912">>,<0.456.0>,<0.472.0>,
                              {db_header,11,0,nil,nil,
                               <<0,0,0,0,161,242,0,0,0,0,19,46>>,
                               1,nil,nil},
                              0,
                              {btree,<0.456.0>,nil,
                               #Fun<couch_db_updater.7.78420415>,
                               #Fun<couch_db_updater.8.78420415>,
                               #Fun<couch_btree.1.39972947>,
                               #Fun<couch_db_updater.9.78420415>,1279,2558,
                               true},
                              {btree,<0.456.0>,nil,
                               #Fun<couch_db_updater.10.78420415>,
                               #Fun<couch_db_updater.11.78420415>,
                               #Fun<couch_db_updater.6.78420415>,
                               #Fun<couch_db_updater.12.78420415>,1279,2558,
                               true},
                              {btree,<0.456.0>,
                               {41458,<<>>,4910},
                               identity,identity,#Fun<couch_btree.1.39972947>,
                               nil,1279,2558,true},
                              0,<<"main/master">>,
                              "/Users/Levon/Library/Application Support/Couchbase/var/lib/couchdb/main/master.couch.2",
                              [],nil,
                              {user_ctx,null,[],undefined},
                              nil,
                              [before_header,after_header,on_file_open],
                              []},
                             <<"de3e8a21744649619e09154819e3a2e2">>,0,
                             undefined,0,0,23,23,0,256547314792210,
                             [2,23],
                             0,
                             {0,0},
                             "Mon, 16 Feb 2015 16:57:50 GMT",
                             {1424107670066990,#Ref<0.0.6.72762>},
                             {1424,105870,66713},
                             {1424,105870,66212},
                             [<0.32530.2>,<0.32531.2>],
                             <0.32527.2>,false,
                             {"http://Administrator:kva%24im0d0@192.168.1.100:9091/",
                              [{<<"vb">>,32},
                               {<<"bucket">>,<<"main_v1">>},
                               {<<"bucketUUID">>,
                                <<"baa2747c512c4817bc52feecde9687ea">>}],
                              {httpdb,
                               "*****@192.168.1.100:9091/main_v1%2f32/",nil,
                               [{"Accept","application/json"},
                                {"User-Agent","CouchDB/2.1.1r-432-gc2af28d"}],
                               180000,
                               [{connect_options,
                                 [{keepalive,true},{nodelay,false}]},
                                {connect_timeout,180000},
                                {send_retry,0}],
                               2,250,xdc_lhttpc_pool,20}}}}]}]
** Reason for termination == 
** {http_request_failed,"POST",
                        "http://Administrator:*****@192.168.1.100:9091/main_v1%2f32/_bulk_docs",
                        {error,{code,500}}}

[error_logger:error,2015-02-16T11:57:50.839,ns_1@couchbase.dev.mydomain.com:error_logger<0.6.0>:ale_error_logger_handler:do_log:203]
=========================CRASH REPORT=========================
  crasher:
    initial call: xdc_vbucket_rep:init/1
    pid: <0.32520.2>
    registered_name: []
    exception exit: {http_request_failed,"POST",
                        "http://Administrator:*****@192.168.1.100:9091/main_v1%2f32/_bulk_docs",
                        {error,{code,500}}}
      in function  gen_server:terminate/6 (gen_server.erl, line 744)
    ancestors: [<0.28880.0>,<0.28876.0>,xdc_replication_sup,xdcr_sup,
                  ns_server_sup,ns_server_cluster_sup,<0.59.0>]
    messages: [{'EXIT',<0.32531.2>,
                      {http_request_failed,"POST",
                          "http://Administrator:*****@192.168.1.100:9091/main_v1%2f32/_bulk_docs",
                          {error,{code,500}}}}]
    links: [<0.28880.0>]
    dictionary: [{changes_reader,<0.32528.2>}]
    trap_exit: true
    status: running
    heap_size: 4185
    stack_size: 27
    reductions: 13798
  neighbours:

[ns_server:debug,2015-02-16T11:58:12.295,ns_1@couchbase.dev.mydomain.com:compaction_new_daemon<0.448.0>:compaction_new_daemon:process_scheduler_message:1288]Starting compaction for the following buckets: 
[<<"main">>,<<"biz">>]
[ns_server:info,2015-02-16T11:58:12.295,ns_1@couchbase.dev.mydomain.com:<0.32665.2>:compaction_new_daemon:spawn_scheduled_kv_compactor:468]Start compaction of vbuckets for bucket main with config: 
[{database_fragmentation_threshold,{30,undefined}},
 {view_fragmentation_threshold,{30,undefined}}]
[ns_server:debug,2015-02-16T11:58:12.296,ns_1@couchbase.dev.mydomain.com:<0.32668.2>:compaction_new_daemon:bucket_needs_compaction:953]`main` data size is 23809, disk size is 2658412
[ns_server:info,2015-02-16T11:58:12.296,ns_1@couchbase.dev.mydomain.com:<0.32669.2>:compaction_new_daemon:spawn_scheduled_kv_compactor:468]Start compaction of vbuckets for bucket biz with config: 
[{database_fragmentation_threshold,{30,undefined}},
 {view_fragmentation_threshold,{30,undefined}}]
[ns_server:debug,2015-02-16T11:58:12.297,ns_1@couchbase.dev.mydomain.com:<0.32672.2>:compaction_new_daemon:bucket_needs_compaction:953]`biz` data size is 13768, disk size is 826375
[ns_server:debug,2015-02-16T11:58:12.297,ns_1@couchbase.dev.mydomain.com:compaction_new_daemon<0.448.0>:compaction_new_daemon:process_compactors_exit:1329]Finished compaction iteration.
[ns_server:debug,2015-02-16T11:58:12.297,ns_1@couchbase.dev.mydomain.com:compaction_new_daemon<0.448.0>:compaction_scheduler:schedule_next:60]Finished compaction too soon. Next run will be in 30s
[ns_server:debug,2015-02-16T11:58:13.241,ns_1@couchbase.dev.mydomain.com:compaction_new_daemon<0.448.0>:compaction_new_daemon:process_scheduler_message:1288]Starting compaction for the following buckets: 
[<<"main">>,<<"biz">>]
[ns_server:info,2015-02-16T11:58:13.242,ns_1@couchbase.dev.mydomain.com:<0.32705.2>:compaction_new_daemon:try_to_cleanup_indexes:564]Cleaning up indexes for bucket `main`
[ns_server:info,2015-02-16T11:58:13.243,ns_1@couchbase.dev.mydomain.com:<0.32705.2>:compaction_new_daemon:spawn_scheduled_views_compactor:494]Start compaction of indexes for bucket main with config: 
[{database_fragmentation_threshold,{30,undefined}},
 {view_fragmentation_threshold,{30,undefined}}]
[ns_server:info,2015-02-16T11:58:13.243,ns_1@couchbase.dev.mydomain.com:<0.32706.2>:compaction_new_daemon:try_to_cleanup_indexes:564]Cleaning up indexes for bucket `biz`
[ns_server:info,2015-02-16T11:58:13.244,ns_1@couchbase.dev.mydomain.com:<0.32706.2>:compaction_new_daemon:spawn_scheduled_views_compactor:494]Start compaction of indexes for bucket biz with config: 
[{database_fragmentation_threshold,{30,undefined}},
 {view_fragmentation_threshold,{30,undefined}}]
[ns_server:debug,2015-02-16T11:58:13.244,ns_1@couchbase.dev.mydomain.com:compaction_new_daemon<0.448.0>:compaction_new_daemon:process_compactors_exit:1329]Finished compaction iteration.
```

Let me know if I should file a new issue for this, 

Thanks, 

Levon

**Comment by ltamrazov** [comment]
Thanks for the fix, it worked. I deleted my previous comment and filed it as a new issue, thought it was more appropriate: #72

**Comment by Branor** [comment]
Fixed in v2.1.1, check the RC here: https://github.com/couchbaselabs/elasticsearch-transport-couchbase/releases/tag/2.1.1-RC
